### ISSUE TITLE
Conversion from Windows 1.x/2.x Icon format not working

### ISSUE THREAD

**Issue by superbonaci** [ImageMagick/ImageMagick#6866]
### ImageMagick version

7.1.1-21

### Operating system

MacOS

### Operating system, version and so on

MacOS 14.1.1

### Description

imagemagick 7.1.1-21 installed with brew.sh
Continuation of #6527
The conversion is not working, it generated 2 black PNGs.

### Steps to Reproduce

```
magick CARDFILE.ICO CARDFILE_imagemagick.png
```

### Images

[CARDFILE-magick.zip](https://github.com/ImageMagick/ImageMagick/files/13368963/CARDFILE-magick.zip)

**Comment by fmw42** [comment]
I get the same result on Mac OSX Ventura with IM 7.1.1.21.  

How do you know that the ico images are not totally black?

**Comment by superbonaci** [comment]
> How do you know that the ico images are not totally black?
They are totally black.

**Comment by fmw42** [comment]
You imply they should not be black?  How do you know that black is not correct?

**Comment by superbonaci** [comment]
This is how CARDFILE.ICO looks like in Windows 1.01:

![](https://user-images.githubusercontent.com/8339596/257310713-86f4c173-e465-49d1-a306-accca4a02cda.png)

**Comment by superbonaci** [comment]
@dlemstra yes let me check if i can compile it from source, I'm currently in
```
Version: ImageMagick 7.1.1-21 Q16-HDRI aarch64 21667 https://imagemagick.org
```

**Comment by superbonaci** [comment]
@dlemstra tried latest [e1ff620](https://github.com/ImageMagick/ImageMagick/commit/e1ff62030d783d4efc24919effaa8b7bf5a84dc3)

`display` and convert to jpg work, but to png doesn't.

![goodico](https://github.com/ImageMagick/ImageMagick/assets/8339596/8b6b351c-c49e-43f1-b84f-74b3e1a9016d)

Icon converted to PNG (not the same as above):

![badicon](https://github.com/ImageMagick/ImageMagick/assets/8339596/5975f946-cdab-43e5-b84f-da060609f17d)

Original files:

![device dependent](https://github.com/python-pillow/Pillow/assets/8339596/96d23c14-7004-4df3-99fc-24f33bddeea5)

[DEV.zip](https://github.com/python-pillow/Pillow/files/12771550/DEV.zip)

**Comment by dlemstra** [comment]
Are you sure the PNG is not "incorrect" because you are at 1600% zoom? I am getting a correct output image for both png and jpg.

**Comment by superbonaci** [comment]
gwenview is displaying it correctly, since the 5 black pixels are the only ones visible considering how each it's represented:
the white at the right is really white color while the white at the left is really transparent. While Imagemagick represents the transparent as grey.

Other apps represent the transparent as alternating white & grey squares so it's noticeable as such:

![kolourpaint-ok](https://github.com/ImageMagick/ImageMagick/assets/8339596/bbfd8200-7683-4480-904c-e929e3963dba)

So basically the issue is fixed, you can close it. Good job.

**Comment by superbonaci** [comment]
@dlemstra is there any way to change the transparent pattern for `display` command? Maybe open another issue?
